File: gc-kettle/kettle-spring-boot-starter/src/main/java/com/gc/kettle/starter/KettleProperties.java
package com.gc.kettle.starter;


import com.gc.kettle.actuator.properties.DatabaseMetaProperties;
import lombok.Getter;
import lombok.Setter;
import org.springframework.boot.context.properties.ConfigurationProperties;

/**
 * kettle 配置参数
 * @author shizhongming
 * 2020/7/16 8:10 下午
 */
@ConfigurationProperties("gc.kettle")
@Getter
@Setter
public class KettleProperties {

    /**
     * 数据库资源库连接池
     */
    private DatabaseMetaProperties repository = new DatabaseMetaProperties();

}

File: smart-starter/kettle-spring-boot-starter/src/main/kotlin/com/smart/starter/kettle/KettleProperties.kt
package com.smart.starter.kettle

import com.smart.kettle.common.config.DatabaseMetaProperties
import org.springframework.boot.context.properties.ConfigurationProperties

/**
 *
 * @author ming
 * 2019/7/11 上午9:43
 */
@ConfigurationProperties(prefix = "smart.kettle")
class KettleProperties {
    /**
     * 资源库配置
     */
    val repository = DatabaseMetaProperties()

}